src/test/java/de/rub/nds/modifiablevariable/integer/ModifiableIntegerTest.java

```diff
@@ -30,94 +26,83 @@ public void setUp() {
+    /** Test of isOriginalValueModified method, of class ModifiableInteger. */
     @Test
     public void testIsOriginalValueModified() {
+        integer1.createRandomModification();
+        assertTrue(integer1.isOriginalValueModified());
+        assertFalse(integer2.isOriginalValueModified());
     }
 
-    /**
-     * Test of getByteArray method, of class ModifiableInteger.
-     */
-    @Disabled("Not yet implemented")
+    /** Test of getByteArray method, of class ModifiableInteger. */
     @Test
     public void testGetByteArray() {
+        assertArrayEquals(integer1.getByteArray(2), integer2.getByteArray(2));
     }
 
-    /**
-     * Test of validateAssertions method, of class ModifiableInteger.
-     */
-    @Disabled("Not yet implemented")
+    /** Test of validateAssertions method, of class ModifiableInteger. */
     @Test
     public void testValidateAssertions() {
+        assertTrue(integer1.validateAssertions());
+        integer1.setOriginalValue(null);
+        assertTrue(integer1.validateAssertions());
     }
 
-    /**
-     * Test of getOriginalValue method, of class ModifiableInteger.
-     */
-    @Disabled("Not yet implemented")
+    /** Test of getOriginalValue method, of class ModifiableInteger. */
     @Test
     public void testGetOriginalValue() {
+        integer1.createRandomModification();
+        assertEquals(integer1.getOriginalValue(), integer2.getOriginalValue());
+        assertNotEquals(integer1.getValue(), integer2.getValue());
     }
 
-    /**
-     * Test of setOriginalValue method, of class ModifiableInteger.
-     */
-    @Disabled("Not yet implemented")
+    /** Test of setOriginalValue method, of class ModifiableInteger. */
     @Test
     public void testSetOriginalValue() {
+        integer2.setOriginalValue(3);
+        assertNotEquals(integer1.getOriginalValue(), integer2.getOriginalValue());
     }
 
-    /**
-     * Test of toString method, of class ModifiableInteger.
-     */
-    @Disabled("Not yet implemented")
+    /** Test of toString method, of class ModifiableInteger. */
     @Test
     public void testToString() {
+        assertEquals(integer1.toString(), integer2.toString());
+        assertNotEquals(integer1.getOriginalValue(), integer1.toString());
     }
 
-    /**
-     * Test of equals method, of class ModifiableInteger.
-     */
+    /** Test of equals method, of class ModifiableInteger. */
     @Test
     public void testEquals() {
         assertEquals(integer1, integer2);
         integer2.setOriginalValue(3);
         assertNotEquals(integer1, integer2);
     }
 
-    /**
-     * Test of hashCode method, of class ModifiableInteger.
-     */
-    @Disabled("Not yet implemented")
+    /** Test of hashCode method, of class ModifiableInteger. */
     @Test
     public void testHashCode() {
+        assertEquals(integer1.hashCode(), integer2.hashCode());
+        integer1.setOriginalValue(4);
+        assertNotEquals(integer1.hashCode(), integer2.hashCode());
```

Comment:
`hashCode` does not mandate that the return values are different if the objects are not equals. It only states that `hashCode()` returns the same value if` `integer1` and `integer2` are equal. Therefore, these two lines should be removed.